### DISCUSSION TITLE
Read access violation.

### DISCUSSION THREAD

**Discussion by xland** [mikke89/RmlUi#406] [Q&A] [answered]
When I close a window in a multi-window application,
I try to release resources with the following code.
```c++
// WindowBase is the base class of all windows
// One window may has multiple documents
WindowBase::~WindowBase() {
	Rml::RemoveContext(windowName);
	Rml::ReleaseTextures(renderInterface);
	delete renderInterface;
	glfwDestroyWindow(glfwWindow);
}
```
VisualStudio prompts me that there is an exception with the following RmlUi code
```c++
// Core\EventDispatcher.cpp
// Detaches all events from this dispatcher and all child dispatchers.
void EventDispatcher::DetachAllEvents()
{
	for (const auto& event : listeners)
		event.listener->OnDetach(element);   // This line has exception
	listeners.clear();
	for (int i = 0; i < element->GetNumChildren(true); ++i)
		element->GetChild(i)->GetEventDispatcher()->DetachAllEvents();
}
```
The exception is:
> Read access violation.
> event.listener-> is 0xFFFFFFFFFFFFFFE7。

The event and element both have value.

What's wrong with my code, Thanks very much!

**Comment by barotto** [comment, marked as answer]
Make sure event listeners attached to any element of documents in the context are kept alive until the call to `Rml::RemoveContext()`.

**Reply by xland** [reply]
Yes!
Problem solved,
Thank you very much!
